src/PIL/IcnsImagePlugin.py

```diff
@@ -302,73 +300,68 @@ def load(self):
-
-    macOS only.
     """
     if hasattr(fp, "flush"):
         fp.flush()
 
-    # create the temporary set of pngs
-    with tempfile.TemporaryDirectory(".iconset") as iconset:
-        provided_images = {
-            im.width: im for im in im.encoderinfo.get("append_images", [])
-        }
-        last_w = None
-        second_path = None
-        for w in [16, 32, 128, 256, 512]:
-            prefix = f"icon_{w}x{w}"
-
-            first_path = os.path.join(iconset, prefix + ".png")
-            if last_w == w:
-                shutil.copyfile(second_path, first_path)
-            else:
-                im_w = provided_images.get(w, im.resize((w, w), Image.LANCZOS))
-                im_w.save(first_path)
-
-            second_path = os.path.join(iconset, prefix + "@2x.png")
-            im_w2 = provided_images.get(w * 2, im.resize((w * 2, w * 2), Image.LANCZOS))
-            im_w2.save(second_path)
-            last_w = w * 2
-
-        # iconutil -c icns -o {} {}
-
-        fp_only = not filename
-        if fp_only:
-            f, filename = tempfile.mkstemp(".icns")
-            os.close(f)
-        convert_cmd = ["iconutil", "-c", "icns", "-o", filename, iconset]
-        convert_proc = subprocess.Popen(
-            convert_cmd, stdout=subprocess.PIPE, stderr=subprocess.DEVNULL
+    # Size
+    sizes = [128, 256, 512, 32, 64, 256, 512, 1024]
+    size_str = ["ic07", "ic08", "ic09", "ic11", "ic12", "ic13", "ic14", "ic10"]
+    file_size = 0
+    entries = []
+    provided_images = {im.width: im for im in im.encoderinfo.get("append_images", [])}
+    temp_sizes = {s: io.BytesIO() for s in set(sizes)}
+    for s, temp in temp_sizes.items():
+        nb = provided_images[s] if s in provided_images else im.resize((s, s))
+        nb.save(temp, "png")
+    for index, s in enumerate(sizes):
+        temp = temp_sizes[s]
+        file_size += len(temp.getvalue())
+        entries.append(
+            {"type": size_str[index], "size": len(temp.getvalue()), "stream": temp}
         )
 
-        convert_proc.stdout.close()
+    # Header
+    fp.write(struct.pack("<i", MAGIC)[::-1])
+    fp.write(struct.pack("<i", file_size)[::-1])
 
-        retcode = convert_proc.wait()
+    # TOC
+    toc_size = HEADERSIZE + (len(entries) * HEADERSIZE)
+    fp.write(struct.pack("<i", _to_int(TOC))[::-1])
+    fp.write(struct.pack("<i", toc_size)[::-1])
+    for e in entries:
+        fp.write(struct.pack("<i", _to_int(e["type"]))[::-1])
+        fp.write(struct.pack("<i", HEADERSIZE + e["size"])[::-1])
 
-        if retcode:
-            raise subprocess.CalledProcessError(retcode, convert_cmd)
+    # Data
+    for index, e in enumerate(entries):
+        fp.write(struct.pack("<i", _to_int(e["type"]))[::-1])
+        fp.write(struct.pack("<i", HEADERSIZE + e["size"])[::-1])
+        fp.write(e["stream"].getvalue())
 
-        if fp_only:
-            with open(filename, "rb") as f:
-                fp.write(f.read())
+    fp.flush()
```

Comment:
```suggestion
    if hasattr(fp, "flush"):
        fp.flush()
```
Mimicking https://github.com/python-pillow/Pillow/pull/4526/files#diff-bb9af648f9d6055f580440e268d715dae9a22a2c4bfd47afd18bd834c05e96d2R317-R318